### DISCUSSION TITLE
Why does /x* match everything in the buffer?

### DISCUSSION THREAD

**Discussion by TheButterflyMon** [vim/vim#14899] [Q&A]
Why does `/x*` or `/\vx*` match everything in the buffer? It should match zero or more `x` characters. Am I missing something?

**Comment by zeertzjq** [comment]
An empty string is zero or more `x` characters, and an empty string matches everything.

**Reply by TheButterflyMon** [reply]
Thank you for responding. By empty string, do you meant an empty string before the characters?

For example, if my buffer contains:

My expectation was that x* should match the xxxx in maxxxxed but nothing in sacked.

Then, running `:%s/\vx*/H/g` produces,

HMHyH HeHpHeHcHtHaHtHiHoHnH HwHaHsH HtHhHaHtH H*H HsHhHoHuHlHdH HmHaHtHcHhH HtHhHeH H HiHnH HmHaHeHdH HbHuHtH HnHoHtHhHiHnHgH HiHnH HsHaHcHkHeHdH.

... and running `:%s/\V\CH//g` produces,

My epectation was that * should match the  in maed but nothing in sacked.

**If this is the way `/x*` works in Vim, is there a way to change the behavior of `*` so that it only matches the xxxx in maxxxxed but nothing in sacked?**

**Reply by chrisbra** [reply]
Well each character matches exactly *zero-times* for `x`

**Reply by EdwardMG** [reply]
> If this is the way /x* works in Vim, is there a way to change the behavior of * so that it only matches the xxxx in maxxxxed but nothing in sacked?

`x\{1,}` and `\vx+` will do what you expect.
`:h pattern-overview` for more details

This is a common tripping point for me too when working with regexp in vim, even though it is the same in most regexp engines for some reason I forget in vim :)

**Reply by benknoble** [reply]
The nomagic variant is `x\+` (easier to type than `x\{1,}`).

**Reply by benknoble** [reply]
I think people forget this in Vim because it's not often that we write such degenerate patterns in programs. Even `grep 'x*' <<<abc` outputs abc, though.

**Reply by TheButterflyMon** [reply]
I understand. Thanks for all your replies.
